fix: form components

Robbert opened the pull request:
<img width="367" alt="Screenshot 2025-02-18 at 00 36 34" src="https://github.com/user-attachments/assets/11864f9f-336a-4ae1-84f9-4574adc194a1" />

Robbert committed:
feat: add `type` to Apache Wicket UtrechtFormLabel

Robbert committed:
build: use `workspace` dependency for design tokens

Robbert committed:
fix: improve checkbox rendering in Apache Wicket

Robbert committed:
refactor: use custom components for SSO example

Robbert committed:
feat: add `setInputType()`

Robbert committed:
fix: improve rendering of Form Field Textbox in Apache Wicket

Robbert committed:
refactor: use NL Design System components for SSO example

keesvandieren committed:
Use add with varargs, fix string comparison

Easier to read when using varargs

compare strings with == or != does not always give correct results (use equals)

keesvandieren committed:
use isEnabled functionality of Wicket

Wicket has it, devs can disable components, also authorization can influence enabledness state (e.g. user is allowed to edit vs user is allowed to read only)

keesvandieren committed:
Add few TODO's

keesvandieren committed:
Better layout

keesvandieren committed:
More efficient and less code

Efficient: behaviors often can be Singletons. If we use singletons for all behaviors where possible it will save hundreds of instantiations per page render.

Less code: using a base class for behaviors to render css makes code per NL Design Component somewhat less

keesvandieren commented:
Ik heb wat verbeteringen in behaviors gepushed, graag ook doorvoeren op  de andere die in form components gebruikt worden.

Commit:
Merge branch 'main' into fix/form-components

# Conflicts:
#	pnpm-lock.yaml

Commit:
#202 ontbrekende CSS-bestanden tijdens het renderen resolven

Het is noodzakelijk dat pnpm gedraaid wordt. Opgenomen instructies.

Het component form-label-css komt niet meer transitief mee. Expliciet
opgenomen.

Commit:
#202 checkbox-component wat beter gemaakt WIP

Rendering is nog niet goed, kennelijk gaat 'utrecht-form-field' niet
langer breedte van de parent overnemen

Commit:
#202 dubbele html markup container eruit gehaald

Daardoor was de rendering niet goed

Commit:
#202 disabled verbeterd, geen dubbele classes

Voor volledige werking is behaviours alleen onvoldoende.

Houden we het statement dat het zonder concrete componenten moet kunnen werken, vast?

Commit:
#202 test checkbox html zoals verwacht

Wordt tegenwoordig in een paragraaf gerenderd
Test met WicketTester dat het klopt

Simpele validatie toegevoegd

Commit:
#202 is-required erbij, description, error erbij

WIP, test faalt nog

Commit:
#202 correct checkbox html and implementation

Correct html is being generated.

storybook has 'is-required' class attribute, but we can drop it:
https://codefornl.slack.com/archives/C01DAT4TRPF/p1757604226842329

Commit:
#202 make checkbox testing util reusable, improve checkbox

Commit:
#202 use models as arguments

Commit:
#202 Textbox basic implementation

Generate correct output in default state

(WIP: error handling, disabled, readonly, autcomplete etc)

Commit:
#202 move error message creation to a factory

it is the same for the 2 form components so should be centralized

Commit:
#202 validatiemelding met de juiste css

Commit:
#202 no longer change tag name

Instead, assert that the developer sets the correct html tag.

Commit:
#202 make nexted bem elements usable out of form fields

In BEM, the '__' represent a nested component.

Add those classes always in context, where they are added as nested
components.

Commit:
#202 introduce interface, describing common requirements

The common requirements are not yet abstracted, we will do this step by
step.

Commit:
#202 move textbox classes to Java

Control everything from Java.

Explain why to have them in Java, not in .html

Commit:
#202 clean up code of Checkbox

Move css classes to where they are used

Use suffix Component to make clear it is a concept from NLDS

Commit:
#202 move checkbox code to java

No CSS classes in HTML for maintainability

Align checkbox and textbox

Commit:
#202 prepare form elements for standalone usage

Commit:
#202 make it look good on SSO page

Enforce usage on a concrete tag. Behavior checks correct tag name

Commit:
#202 move some label enabledness to behavior

If the form is disabled (or the form field component) Behavior manages.

If form control is disabled, UtrechtFormField will take that into
account.

Commit:
#202 make components implement interface

Do basic verifications based on the interface

Move more classes to constants

Commit:
119 checkbox loskoppelen van form field

Commit:
#119 checkbox loskoppelen van form field

Moet ook standalone te gebruiken zijn, bijvoorbeeld als filter in een
tabel.

Naamgeving beter gemaakt: Container > FormField zodat onderscheid beter
is.

Eenvoudige test toegevoegd voor controle standaard output

Storybook generator toegevoegd

Commit:
#119 make checkbox look a tiny bit better

Commit:
#119 make storybook with multiline statements

Commit:
#119 checkbox css is loaded by the Checkbox, not Form field

Commit:
#119 drop prebuild, breaks build on M4 Pro

Recommended as solution for this issue by Bryan de Jong

See https://codefornl.slack.com/archives/C01DAT4TRPF/p1758269924550059

Commit:
#119 remove duplicate (fixes warning while running storybook)

Commit:
#119 correct grammar: clas > class

Commit:
#119 use chaining for better readability

Commit:
#119 do not pass nullable components, refactor

Instead, subclass where the components are needed

Commit:
#119 use new instead of create prefix

That is the way Wicket works

Commit:
#119 requires doctype and title for sonar

Commit:
#119 use wicket:label instead of custom container

Checkbox now has a label, can be used in validation messages too.

Commit:
#119 better names

Commit:
#119 centralize wicket tag assert

Commit:
#119 code cleanup

Commit:
#119 Textbox and UtrechtFormFieldTextbox should be seperated

Commit:
#119 sonar warning

Commit:
#123 extract textbox from form field

Makes it usable in a standalone way

Commit:
#123 reduce visibility of classes

We want to keep implementation details private

Commit:
#123 Fix some sonar warnings (WOT)

Commit:
#123 null value are allowed. Make this explicit

Commit:
#123 centralize class name attribute for behaviors

Mainly to make sonar happy.

Commit:
#123 consistent css class names

Commit:
#123 mark component stability

Commit:
#123 append css classes for ComponentTag

Centralize logic to have literal 'class' once.

Commit:
#123 fix typo

Commit:
#123 Css constants to enums, utilize enums.

Instead of a list of constants, make them an enum.

Move some utility methods to the enum.

Commit:
#123 #250 disable wicket html5, correct validation bug

Add extra validator.

Error message view bind to text field, not to its container

Commit:
#119 'for' attribute is still needed in Java code

The wicket:label inserts the label (text) from the input component.

It cannot do the markup as label needs NL Design System style classes.
We want all classes to be added by java code

Commit:
#123 use wicket:label to get the text

Stil needs a WebMarkupContainer to set the css classes

Commit:
#202 mark component states

This makes it easier to indicate if a component is ready for use in
production applications.

Commit:
#202 delete unused imports

Commit:
#202 shortcut and makes list unmodifiable

Commit:
#202 explain why deprecated

Commit:
#202 some sonar fixes
